packages/apex-node/src/tests/asyncTests.ts

```diff
@@ -65,14 +68,17 @@ export class AsyncTests {
           sClient.disconnect();
         });
 
-      const asyncRunResult = await sClient.subscribe(
-        this.getTestRunRequestAction(options)
-      );
+      const testRunId = await this.getTestRunRequestAction(options)();
+
+      if (exitEarly) {
+        return { testRunId };
+      }
 
       if (token && token.isCancellationRequested) {
         return null;
       }
 
+      const asyncRunResult = await sClient.subscribe(undefined, testRunId);
```

Comment:
Is the first parameter still necessary after this change? Also the `setInterval` calls for the streaming error fix needs to be adjusted for this change.

Reply:
I was thinking we can leave the parameter in for now and if things look good & settle down after the recent streaming fixes then we can remove the first parameter as an option

Reply:
@xyc how would you recommend changing the `setInterval` calls for the streaming error fix? `setInterval` will never be called right now so I'm trying to understand where the `if (!this.hasDisconnected)` block should be moved.... potentially [here](https://github.com/forcedotcom/salesforcedx-apex/blob/c0adfc5ce28c0df7c01146013595950ab64a8c17/packages/apex-node/src/streaming/streamingClient.ts#L211)? But I'm not sure it will run in the correct order since it's no longer in a then block

Reply:
yeah that should be the place, let's copy / paste the `if (!this.hasDisconnected)` block since you mentioned you'll keep the param and both conditions at the moment.